task/src/main/java/com/flipkart/flux/impl/task/TaskExecutor.java

```diff
@@ -78,7 +78,12 @@ protected Event run() throws Exception {
 		}
 		final Object returnObject = result.getKey();
 		if (returnObject != null) {
-			return new Event(outputeEventName, returnObject.getClass().getCanonicalName(), Event.EventStatus.triggered, stateMachineId, objectMapper.writeValueAsString(returnObject), MANAGED_RUNTIME);
+            Event event = (Event) returnObject;
+            event.setName(outputeEventName);
```

Comment:
create another object/abstraction :) You can return a class called SerialisedEventData , or simply a string.
Use that to create the Event object in task executor :)